Tests/LibWeb/add_libweb_test.py

```diff
@@ -9,52 +9,124 @@
 TEST_DIR = Path(__file__).resolve().parent
 
 
-def create_text_test(test_name: str, is_async: bool = False) -> None:
+def create_test(test_name: str, test_type: str, is_async: bool = False) -> None:
     """
-    Create a new Text type test file with the given test name.
+    Create a test of a given type with the given test name
 
     Args:
         test_name (str): Name of the test.
+        test_type (str): Type of the test. Currently supports
+            "Text", "Layout", "Screenshot" and "Ref""
         is_async (bool, optional): Whether it is an async test. Defaults to False.
     """
-    input_prefix = TEST_DIR / "Text" / "input" / test_name
-    input_dir = input_prefix.parent
+
+    if test_type in ["Layout", "Text"]:
+        input_prefix = TEST_DIR / test_type / "input" / test_name
+    else:
+        input_prefix = TEST_DIR / test_type / test_name
     input_file = input_prefix.with_suffix(".html")
+    input_dir = input_prefix.parent
 
-    expected_prefix = TEST_DIR / "Text" / "expected" / test_name
-    expected_dir = expected_prefix.parent
-    expected_file = expected_prefix.with_suffix(".txt")
+    if test_type in ["Layout", "Text"]:
+        output_prefix = TEST_DIR / test_type / "expected" / test_name
+        output_file = output_prefix.with_suffix(".txt")
+    else:
+        output_prefix = TEST_DIR / test_type / "reference" / test_name
+        output_file = output_prefix.with_suffix(".html")
+    output_dir = output_prefix.parent
 
     # Create directories if they don't exist
     input_dir.mkdir(parents=True, exist_ok=True)
-    expected_dir.mkdir(parents=True, exist_ok=True)
+    output_dir.mkdir(parents=True, exist_ok=True)
 
     num_sub_levels = len(Path(test_name).parents) - 1
     path_to_include_js = "../" * num_sub_levels + "include.js"
 
-    # Create input and expected files
-    input_file.write_text(fR"""<!DOCTYPE html>
+    def generate_boilerplate():
+        """
+        return a tuple of strings with the input and
+        output boilerplate for a given test type
+        """
+
+        generic_boilerplate = R"""<!DOCTYPE html>
+<head>
+<style>
+</style>
+</head>
+<body>
+</body>
+"""
+
+        if test_type == "Text":
+            input_boilerplate = fR"""<!DOCTYPE html>
 <script src="{path_to_include_js}"></script>
 <script>
     {f"asyncTest(async (done)" if is_async else "test(()"} => {{
         println("Expected println() output");
     {f"    done();" if is_async else ""}
     }});
 </script>
-""")
-    expected_file.write_text("Expected println() output\n")
+"""
+            expected_boilerplate = "Expected println() output\n"
+
+        elif test_type == "Ref":
+            input_boilerplate = fR"""<!DOCTYPE html>
+<head>
+     <link rel="match" href="reference/{test_name}-ref.html" />
```

Comment:
Found a bug in this - `test_name` includes the file extension, so you get `href="reference/cool-test.html-ref.html"` instead of `href="reference/cool-test-ref.html"`

Reply:
Right, so I guess this will depend on what the user puts in at the command line. If they do `cool-test.html` then you'll get the double extension, but `cool-test` will be fine. Right now this works under the assumption there's no file extension. We could assert that `test_name` has no periods or something, but that's a bit hacky. If you have a more robust solution I could do that too

Reply:
Oh interesting, the other test types work OK with that input because `with_suffix()` doesn't add a suffix if it's already there. I think you could make use of [`stem`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem) to remove the suffix from `test_name` at the start of `create_test()` maybe? Otherwise, just complaining if the path ends with `.html` would do - this script can't create tests with other extensions anyway.